### ISSUE TITLE
Legend entries linked to the same layer disappear on reload

### ISSUE THREAD

**Issue by AleksueiR** [fgpv-vpgf/fgpv-vpgf#2901]
### Problem

When several legend entries point to the same layer, only the first of these legend entries is restored when the layers is reloaded. Observe:

![](https://i.imgur.com/crDzXSY.gif)

It seems the problem is in here: https://github.com/fgpv-vpgf/fgpv-vpgf/blob/develop/src/app/geo/legend.service.js#L279
There is only one promise created for all the legend entries removed on reload and it will re-insert the entry back only once. All other removed entries will not be restored.

### Version
2.4.0 [45ea02a88]
2018-07-27 12:13:09

**Comment by AleksueiR** [comment]
Almost a duplicate of https://github.com/fgpv-vpgf/fgpv-vpgf/issues/2657.

**Comment by ShrutiVellanki** [comment]
Can confirm was included in release `v2.5.0`  
- checked out the `v2.5.0` tag
- the changes from https://github.com/fgpv-vpgf/fgpv-vpgf/pull/2919 were included in the code

Still no longer an issue as checked with [current develop](http://fgpv.cloudapp.net/demo/develop/dev/samples/index-samples.html)

Closing issue.
